### PR TITLE
Added toolStripFileExplorer to complete the feature-request

### LINKED ISSUE #3585: Suggestion: Add a "File Explorer" shortcut in the toolbar (next to "Git bash")
Just noticed recently it exists in the Repository menu, but thought it would deserve its own spot in the toolbar next to "Git bash" as people often want to open the repository root in explorer.

### PR DESCRIPTION
**Pull request by joshia89**
Fixes #3585  .

Changes proposed in this pull request:
 - Add a "File Explorer" shortcut in the toolbar (next to "Git bash")
 
Screenshots before and after (if PR changes UI):
- 
image

**How did I test this code:**
 - Make sure the button appears correctly
 - If it is a valid working directory, the file explorer is enabled and behaves as expected
 - If it is not a valid working directory, the file explorer button is disabled

### CODE REVIEW

File: GitUI/CommandsDialogs/FormBrowse.cs [hunk 1]
```diff
@@ -1902,6 +1903,18 @@ private void ToolStripButtonPushClick(object sender, EventArgs e)
             PushToolStripMenuItemClick(sender, e);
         }
 
+        private void ToolStripFileExplorerClick(object sender, EventArgs e)
```
Comment (RussKie) [inline GitUI/CommandsDialogs/FormBrowse.cs]: there is already a handler doing just this - `FileExplorerToolStripMenuItemClick`
please bind the event to the existing handler

File: GitUI/CommandsDialogs/FormBrowse.Designer.cs [hunk 2]
```diff
@@ -528,6 +530,17 @@ private void InitializeComponent()
...
             // 
+            // toolStripFileExplorer
+            // 
+            this.toolStripFileExplorer.CheckOnClick = true;
+            this.toolStripFileExplorer.Enabled = false;
+            this.toolStripFileExplorer.Image = global::GitUI.Properties.Resources.Folder;
+            this.toolStripFileExplorer.ImageTransparentColor = System.Drawing.Color.Gray;
+            this.toolStripFileExplorer.Name = "toolStripFileExplorer";
+            this.toolStripFileExplorer.Size = new System.Drawing.Size(23, 22);
+            this.toolStripFileExplorer.ToolTipText = "File Explorer";
```
Comment (RussKie) [inline GitUI/CommandsDialogs/FormBrowse.Designer.cs]: this text needs to be translated

the easiest way is probably do the following:

```
        private FormBrowse()
        {
            InitializeComponent();
            Translate();

            toolStripFileExplorer.ToolTipText = fileExplorerToolStripMenuItem.Text;
        }
```

@jbialobr what do you think?

Comment (jbialobr) [reply GitUI/CommandsDialogs/FormBrowse.Designer.cs]: ToolTipTexts assigned this way are translated automagically. https://github.com/gitextensions/gitextensions/issues/3745

Comment (RussKie) [reply GitUI/CommandsDialogs/FormBrowse.Designer.cs]: For the automagic to work, someone will still need to provide translations, won't they?
`fileExplorerToolStripMenuItem` performs exactly the same function and has the required text, which is already translated. Hence my suggestion to copy and assign the existing translation to the new element.

...Or does the automagic work somehow differently?

Comment (jbialobr) [reply GitUI/CommandsDialogs/FormBrowse.Designer.cs]: >For the automagic to work, someone will still need to provide translations, won't they?

Yes, that is true. But I would not mix `Text` and `ToolTipText`. It so happened they share the same text, but it should be handled on the translation side to prompt the same translation for the same texts when a translator is translating.

Comment (joshia89) [reply GitUI/CommandsDialogs/FormBrowse.Designer.cs]: Thanks @RussKie and @jbialobr.  RussKie, do you want me to make any changes here? 

P.S the other requested change has been committed

**Comment by RussKie** [comment]
no worries then

**Pull request merged**